### PR TITLE
Forcing intel-mpi to use verbs

### PR DESCRIPTION
**Pull request by nrichart**

### CODE REVIEW

File: stacks/syrah/templates/modules.yaml.j2 [hunk 1]
```diff
@@ -169,6 +169,7 @@ modules:
           set:
             I_MPI_PMI_LIBRARY: /usr/lib64/libpmi2.so
             I_MPI_EXTRA_FILESYSTEM: '0'
+            FI_PROVIDER: verbs
```
Comment (batatasfritas) [inline stacks/syrah/templates/modules.yaml.j2:172]: The currently accepted best practice is to use the form:
```suggestion
            I_MPI_OFI_PROVIDER: verbs
```

even though both work. While trying to find proof of this, I've found one reference dating to Intel MPI 2019, perhaps earlier ones exist, but it definitely works for the Intel MPI 2021 we have:
https://www.intel.com/content/www/us/en/developer/articles/technical/mpi-library-2019-over-libfabric.html#inpage-nav-3-undefined

**Review by batatasfritas — COMMENTED** [review]
Since both work, feel free to ignore the comment I just made. But for newer stacks we should opt for the latter...

File: stacks/syrah/templates/modules.yaml.j2 (hunk 1 above)
Comment (nrichart) [reply stacks/syrah/templates/modules.yaml.j2:172]: I also read the documentation for version 2020 that suggest I_MPI_OFI_PROVIDER but newer documentation do not mention this variable anymore
https://www.intel.com/content/www/us/en/docs/mpi-library/developer-guide-linux/2021-6/ofi-providers-support.html

Comment (nrichart) [reply stacks/syrah/templates/modules.yaml.j2:172]: 2021.6 is the version we have installed

**Pull request merged**